<?xml version="1.0"?>
<launch>
  <!-- Configure inputs -->
  <arg name="node_namespace" default="apriltag_ros_node"/>
  <arg name="camera_name" default="/puma/sensors/camera_front/color"/>
  <arg name="camera_frame" default="camera_link"/>
  <arg name="image_topic" default="image_raw"/>
  <!-- Set params -->
  <rosparam command="load" file="$(find puma_tag_detector)/config/settings.yaml" ns="$(arg node_namespace)"/>
  <rosparam command="load" file="$(find puma_tag_detector)/config/tags.yaml" ns="$(arg node_namespace)"/>
  <!-- run node -->
  <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="$(arg node_namespace)" clear_params="true">
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />
    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="false" /> 
  </node>
</launch>